File: files/400-499/437/CH-437.py
import pandas as pd

path = "400-499/437/CH-437 Custom Grouping.xlsx"
input = pd.read_excel(path, usecols="B:D", nrows=12, skiprows=2)
test = pd.read_excel(path, usecols="F:G", nrows=5, skiprows=2)

input["sorted_products"] = (
    input["Products"]
    .str.split(",")
    .apply(lambda x: ",".join(sorted(p.strip() for p in x)))
)
aggregated = (
    input.groupby("sorted_products", as_index=False, sort=False)["Quantity"]
    .sum()
    .rename(columns={"Quantity": "Quantity_sum"})
)
aggregated.columns = test.columns
print(aggregated.equals(test))
# BMD should be BDM

File: files/400-499/437/CH-437.R
library(tidyverse)
library(readxl)

path <- "400-499/437/CH-437 Custom Grouping.xlsx"
input <- read_excel(path, range = "B3:D14")
test <- read_excel(path, range = "F3:G8")

result <- input %>%
  mutate(
    sorted_products = str_split(Products, ",") %>%
      map(~ str_c(sort(.), collapse = ","))
  ) %>%
  unnest() %>%
  summarise(total = sum(Quantity, na.rm = TRUE), .by = sorted_products)
names(result) <- names(test)

all.equal(result, test)
# BMD should be BDM
